### PR TITLE
[ci] Add script to get the build ID for cached bitstreams

### PR DESCRIPTION
**Pull request by a-will**
The old method of checking the base directory no longer works. Get the build ID from the manifest directly.

### CODE REVIEW

**Comment by a-will** [comment]
This was a quick fix for a bug, but I guess I've also now laid the groundwork for using cached builds for the hyperdebug bitstream. Good side effect? ;)

File: ci/scripts/get-bitstream-strategy.sh [hunk 1]
```diff
@@ -33,11 +33,15 @@ else
...
 # Retrieve the most recent bitstream at or before HEAD.
-BITSTREAM="--refresh HEAD" ./bazelisk.sh build ${bitstream_target}
+BITSTREAM="--refresh HEAD" ./bazelisk.sh build @bitstreams//:manifest
 
 # The directory containing the bitstream is named after the git hash.
-bitstream_file=$(ci/scripts/target-location.sh ${bitstream_target})
-bitstream_commit=$(basename "$(dirname ${bitstream_file})")
+manifest_file=$(ci/scripts/target-location.sh @bitstreams//:manifest)
+
+bitstream_commit=$(./util/py/scripts/get_bitstream_build_id.py \
```
Comment (a-will) [inline ci/scripts/get-bitstream-strategy.sh:41]: Hm... might want to consider catching exceptions here. If the design doesn't exist in the cache for some reason, we should proceed to simply building it.

Comment (a-will) [reply ci/scripts/get-bitstream-strategy.sh:41]: Done. Now an empty `bitstream_commit` signifies that the design is not in the cache. Other errors will still trigger build failure.

File: ci/scripts/get-bitstream-strategy.sh [hunk 2]
```diff
@@ -8,8 +8,8 @@ set -e
 usage_string="
 Usage: get-bitstream-strategy.sh <cached-bitstream-target> [pathspec]...
 
-    cached-bitsream-target The bazel target for the cached bitstream. This script
-                           will retrieve the most recent image from a commit in this
+    cached-bitsream-design The design name for the cached bitstream. This script
```
Comment (andreaskurth) [inline ci/scripts/get-bitstream-strategy.sh:11]: Typo:
```suggestion
    cached-bitstream-design The design name for the cached bitstream. This script
```

File: ci/scripts/get-bitstream-strategy.sh [hunk 3]
```diff
@@ -8,8 +8,8 @@ set -e
 usage_string="
 Usage: get-bitstream-strategy.sh <cached-bitstream-target> [pathspec]...
```
Comment (andreaskurth) [inline ci/scripts/get-bitstream-strategy.sh:9]:
```suggestion
Usage: get-bitstream-strategy.sh <cached-bitstream-design> [pathspec]...
```

File: ci/scripts/get-bitstream-strategy.sh [hunk 4]
```diff
@@ -33,22 +33,31 @@ else
...
-echo "Checking for changes against pre-built bitstream from ${bitstream_commit}"
-echo "Files changed:"
-git diff --stat --name-only ${bitstream_commit}
-echo
-echo "Changed files after exclusions applied:"
-# Use the cached bitstream if no changed files remain.
-if git diff --exit-code --stat --name-only ${bitstream_commit} -- "${excluded_files[@]}"; then
-  bitstream_strategy=cached
-else
+bitstream_commit=$(./util/py/scripts/get_bitstream_build_id.py \
```
Comment (andreaskurth) [inline ci/scripts/get-bitstream-strategy.sh:41]: Shouldn't `util/py/scripts/get_bitstream_build_id.py` be invoked through Bazel? Or why are we adding it to `util/py/scripts/BUILD`?

**Review by andreaskurth — APPROVED** [review]
LGTM overall, thanks @a-will! I only have one question and two minor suggestions

File: ci/scripts/get-bitstream-strategy.sh (hunk 4 above)
Comment (a-will) [reply ci/scripts/get-bitstream-strategy.sh:41]: Yeah, it should. I bailed on that when I noticed bazel hiding access to workspace files that weren't in srcs, deps, or data. It was adding `@bitstreams//:manifest` that I felt unsure about, but on second thought, that ought to be fine.

File: ci/scripts/get-bitstream-strategy.sh (hunk 2 above)
Comment (a-will) [reply ci/scripts/get-bitstream-strategy.sh]: Done

File: ci/scripts/get-bitstream-strategy.sh (hunk 4 above)
Comment (a-will) [reply ci/scripts/get-bitstream-strategy.sh]: Updated to run from bazel. Thanks for poking me to "do it right" :)

**Pull request merged**